packages/relay/src/lib/debug.ts

```diff
@@ -106,25 +107,35 @@ export class DebugImpl implements Debug {
   @rpcMethod
   @rpcParamValidationRules({
     0: { type: ParamType.TRANSACTION_HASH_OR_ID, required: true },
-    1: { type: ParamType.COMBINED_TRACER_TYPE, required: false },
-    2: { type: ParamType.TRACER_CONFIG, required: false },
+    1: { type: ParamType.TRACER_CONFIG_WRAPPER, required: false },
   })
+  @rpcParamLayoutConfig(RPC_LAYOUT.custom((params) => [params[0], params[1]]))
   async traceTransaction(
     transactionIdOrHash: string,
-    tracer: TracerType,
-    tracerConfig: ITracerConfig,
+    tracerObject: TransactionTracerConfig,
     requestDetails: RequestDetails,
   ): Promise<any> {
     if (this.logger.isLevelEnabled('trace')) {
       this.logger.trace(`${requestDetails.formattedRequestId} traceTransaction(${transactionIdOrHash})`);
     }
+
+    // we currently do not support prestate tracer
+    if (tracerObject?.tracer === TracerType.PrestateTracer) {
+      throw predefined.INTERNAL_ERROR('Prestate tracer is not yet supported on debug_traceTransaction');
+    }
+
+    //we use a wrapper since we accept a transaction where a second param with tracer/tracerConfig may not be provided
+    //and we will still default to opcodeLogger
+    const wrapper = tracerObject ?? {};
+    const tracer = wrapper.tracer ?? TracerType.OpcodeLogger;
+    const tracerConfig = wrapper.tracerConfig ?? {};
+
```

Comment:
Below should work, right?

```suggestion
const tracer = tracerObject?.tracer ?? TracerType.OpcodeLogger;
const tracerConfig = tracerObject?.tracerConfig ?? {};
```

Reply:
I would assume it works yes, it is essentially the same without the extra variable